service/worker/scheduler/spec.go

```diff
@@ -103,9 +104,12 @@ func (cs *compiledSpec) getNextTime(after time.Time) (nominal, next time.Time, h
 		after = nominal
 	}
 
+	maxJitter := timestamp.DurationValue(cs.spec.Jitter)
 	// Ensure that jitter doesn't push this time past the _next_ nominal start time
-	following := cs.rawNextTime(nominal)
-	next = cs.addJitter(nominal, following.Sub(nominal))
+	if following := cs.rawNextTime(nominal); !following.IsZero() {
+		maxJitter = common.MinDuration(maxJitter, following.Sub(nominal))
```

Comment:
If jitter is following.Sub(nominal), it will push the run to its next run time, will that result in one skip?

Reply:
If you set jitter high enough so that it might go past the next run, so that this limits it to the next run, and then by chance it comes out very close to the end, and your overlap policy is skip, then yes, it would be skipped. That's sort of "user error".. you should either set your jitter to less than the next run + expected run time, or else use BufferOne/All

Reply:
since we handle the jitter over next run here already, I think we should also handle the case where it reaches next run as well. So maxJitter should be the min of maxJitter and following.Sub(nominal)-1 (nanosecond maybe?)

Reply:
The calculation in `addJitter` already effectively subtracts 1 millisecond from the max, it never adds exactly `maxJitter`